mango-python/bdgenomics/mango/QC.py

```diff
@@ -219,3 +221,136 @@ def plot(self, xScaleLog = False, yScaleLog = False, testMode = False, plotType=
             plt.show()
 
         return chromosomes
+
+class VariantEffectAlleleFreq(object):
+    """
+    todo: this needs to be modified to calculate bins in Spark so as to reduce Collect size
```

Comment:
What is this TODO for?

Reply:
In my current implementation the `collect` that is taking place is bringing every variant and its frequency into driver and into Matplotlib.
This seems to me to be using Spark quite poorly, but rather we should predefine the histogram bins, and use a a Reduce in Spark to count the members of each bin and find a way to plot on the driver based on the summarized bin counts  rather than returning so much data to the driver.